app/javascript/packs/Onboarding.jsx

```diff
@@ -0,0 +1,43 @@
+import { h, render } from 'preact';
+import { getUserDataAndCsrfToken } from '../chat/util';
+import getUnopenedChannels from '../src/utils/getUnopenedChannels';
+
+HTMLDocument.prototype.ready = new Promise(resolve => {
+  if (document.readyState !== 'loading') {
+    return resolve();
+  }
+  document.addEventListener('DOMContentLoaded', () => resolve());
+  return null;
+});
+
+function isUserSignedIn() {
+  return (
+    document.head.querySelector(
+      'meta[name="user-signed-in"][content="true"]',
+    ) !== null
+  );
+}
+
+function renderPage() {
+  import('../src/Onboarding')
```

Comment:
@benhalpern, you'll like this performance tweak. We only load onboarding if the user is signed in and hasn't previously seen the onboarding. Before, the component was imported all the time.

Reply:
I do. I'm also excited to improve on several fronts here.

Reply:
This is great!